internal/cluster/distribution/eks/eksworkflow/activity_update_node_group.go

```diff
@@ -69,7 +69,7 @@ type UpdateNodeGroupActivityInput struct {
 
 	ClusterTags map[string]string
 
-	TemplateVersion string
+	CurrentTemplateVersion semver.Version
```

Comment:
I can't recall what `semver.Version` is: is it a plain old Go struct with exported fields? Asking because serializability is usually an issue.

Reply:
No, [it](https://github.com/Masterminds/semver/blob/master/version.go#L47) is not.
According to my basic [test](https://play.golang.org/p/iDz0m5uoh1Q) string and JSON marshaling works fine, however YAML marshaling does not.
Also at the stack interface it is treated as basic string (which works fine).

I picked this library because this is the most common semantic version library we are [using](https://github.com/search?q=org%3Abanzaicloud+github.com%2FMasterminds%2Fsemver) through banzaicloud, so I thought it passes the necessary requirements.

If we wish, I can change this back to a simple string, but then we have to manually construct semver.Version from it for version checking at every use case.
Alternatively I could define an internal type for this which implements YAML marshaling as expected on top of the existing third party type.

What approach would we prefer here?

Reply:
The reason it works with JSON is (obviously) because it implements the marshaler interfaces: https://github.com/Masterminds/semver/blob/7bb0c843b53d6ad21a3f619cb22c4b442bb3ef3e/version.go#L415-L437

I don't think we'll be changing the default serialization layer any time soon (although if/when we switch to temporal, this could be an issue if they use a different serialization mechanism), but the downside is this is a hidden source of potential bugs. If we stick to this solution, I'd probably mark it somehow (with a searchable comment?), so we have a better understanding of the situation if we ever get to a point where we need to make some changes.

The string solution doesn't have this kind of issue, but at the same time it allows passing invalid data to the activity, resulting in a potential failure in an async workflow.

Both solutions have their downsides, I leave the decision to you.

Reply:
Fixed in https://github.com/banzaicloud/pipeline/pull/3357/commits/e177bc75c5248d64bcd105857884a046db21e3cb, decided to roll string based internal implementaiton and updated the usage in https://github.com/banzaicloud/pipeline/pull/3357/commits/3d96e45861c8565f8d66c104213a665da8000301 and https://github.com/banzaicloud/pipeline/pull/3357/commits/2de1ced9a2933b4b60d1840b4d50b3f126be6ea0.